### PR TITLE
Add support for symbol graphs and access notes

### PR DESCRIPTION
**Pull request by beccadax**
This commit adds support for the `-access-notes-path`, `-emit-symbol-graph`, and `-emit-symbol-graph-dir` flags to the new driver.

I have also updated Options.swift; I *think* I've handled all of the new options that can be passed into the driver, but I'd like a second pair of eyes to confirm that.

Note that the symbol graph tests still don't pass because of an issue with the frontend's handling of the `-emit-symbol-graph-dir` flag; I've filed rdar://74418885 about that.

No new tests because I couldn't find anywhere that similar passthrough flags were being tested in the driver. If I've missed something, please let me know.

### CODE REVIEW

**Comment by beccadax** [comment]
@swift-ci please test

File: Sources/makeOptions/makeOptions.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@ enum SwiftFlags {
   ArgumentIsPath = (1 << 12),
   ModuleInterfaceOption = (1 << 13),
   SupplementaryOutput = (1 << 14),
+  SwiftAPIExtractOption = (1 << 15),
```
Comment (cltnschlosser) [inline Sources/makeOptions/makeOptions.cpp:62]: What's this do?

**Review by cltnschlosser — APPROVED** [review]
LGTM

**Comment by cltnschlosser** [comment]
> I have also updated Options.swift; I think I've handled all of the new options that can be passed into the driver, but I'd like a second pair of eyes to confirm that.

You used the makeOptions thing to generate the Swift file, right? It looks like you did.

File: Sources/makeOptions/makeOptions.cpp (hunk 1 above)
Comment (beccadax) [reply Sources/makeOptions/makeOptions.cpp:62]: It was recently added to the Swift compiler in apple/swift#35690; the only reason we care about it is that it's used in Options.td, so we need to support it in the `makeOptions` tool.

**Comment by QuietMisdreavus** [comment]
Paging @bitjammer who implemented `-emit-symbol-graph[-dir]`

**Review by QuietMisdreavus — APPROVED** [review]
Symbol graph flags look fine, but i'll also defer to @bitjammer if she has anything else to add.

**Review by artemcm — APPROVED** [review]

**Pull request merged**